agent/taskresource/asmauth/asmauthstatus.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package asmauth
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+)
+
+type ASMAuthStatus taskresource.ResourceStatus
+
+const (
+	// is the zero state of a task resource
+	ASMAuthStatusNone ASMAuthStatus = iota
+	// represents a task resource which has been created
+	ASMAuthStatusCreated
+	// represents a task resource which has been cleaned up
+	ASMAuthStatusRemoved
+)
+
+var asmAuthStatusMap = map[string]ASMAuthStatus{
+	"NONE":    ASMAuthStatusNone,
+	"CREATED": ASMAuthStatusCreated,
+	"REMOVED": ASMAuthStatusRemoved,
+}
+
+// StatusString returns a human readable string representation of this object
+func (as ASMAuthStatus) String() string {
+	for k, v := range asmAuthStatusMap {
+		if v == as {
+			return k
+		}
+	}
+	return "NONE"
+}
+
+// MarshalJSON overrides the logic for JSON-encoding the ResourceStatus type
+func (as *ASMAuthStatus) MarshalJSON() ([]byte, error) {
+	if as == nil {
```

Comment:
in taskresource/asmauth, this is treated as an error. Here, it's not. Any reason for the differing behavior here?

Reply:
ah that's a miss, will also treat as error.

Reply:
also yay @ now can be an entry in the MarhsalJSON test table.